src/system/python_loader.cc

```diff
@@ -77,45 +79,9 @@ PyObject * PythonLoader::load_module_from_file(const std::string& fname) {
 
 PyObject * PythonLoader::load_module_from_string(const std::string& module_name, const std::string& source_string) {
     initialize();
-
-    // for unknown reason module name is non-const, so we have to make char * copy
-    char * tmp_name = new char[ module_name.size() + 2 ];
-    strcpy( tmp_name, module_name.c_str() );
-
-    PyObject * compiled_string = Py_CompileString( source_string.c_str(), module_name.c_str(), Py_file_input );
-    if (PyErr_Occurred()) {
-    	PyObject *ptype, *pvalue, *ptraceback, *pystr;
-		PyErr_Fetch(&ptype, &pvalue, &ptraceback);
-
-		if ( !PyString_Check(pvalue) ) { // syntax error
-			stringstream ss;
-			PyObject* err_type = PySequence_GetItem(pvalue, 0);
-			pystr = PyObject_Str(err_type);
-			ss << PyString_AsString(pystr) << endl;
-			PyObject* descr = PySequence_GetItem(pvalue, 1);
-			PyObject* file = PySequence_GetItem(descr, 0);
-			pystr = PyObject_Str(file);
-			ss << "  File \"" << PyString_AsString(pystr) << "\"";
-			PyObject* row = PySequence_GetItem(descr, 1);
-			pystr = PyObject_Str(row);
-			ss << ", line " << PyString_AsString(pystr) << endl;
-			PyObject* line = PySequence_GetItem(descr, 3);
-			pystr = PyObject_Str(line);
-			ss << PyString_AsString(pystr);
-			PyObject* col = PySequence_GetItem(descr, 2);
-			pystr = PyObject_Str(col);
-			int pos = atoi( PyString_AsString(pystr) );
-			ss << setw(pos) << "^" << endl;
-			THROW(ExcPythonError() << EI_PythonMessage( ss.str() ));
-    	} else {
-    		THROW(ExcPythonError() << EI_PythonMessage( PyString_AsString(pvalue) ));
-    	}
-    }
-
-    PyObject * result = PyImport_ExecCodeModule(tmp_name, compiled_string);
+    PyObject * compiled_string = Py_CompileString(source_string.c_str(), module_name.c_str(), Py_file_input);
+    PyObject * result = PyImport_ExecCodeModule(module_name.c_str(), compiled_string);
```

Comment:
We should keep dedicated error checking code. It is not simply replaceable by check_error, since here we check success of compilation. Do not remove it unless you compare results of this one and check_error.